components/common-go/namegen/workspaceid.go

```diff
@@ -15,7 +15,13 @@ import (
 
 // WorkspaceIDPattern is the expected Worksapce ID pattern
 // gitpod-protocol/src/util/generate-workspace-id.ts is authoritative over the generation
-var WorkspaceIDPattern = regexp.MustCompile(`^[a-z]{3,12}-[a-z]{2,16}-[a-z0-9]{11}$`)
+
+/*
+	 Regex Pattern Description:
+											Repo. Owner  	Repo. Name  	Random
+											2-16 chars   	2-11 chars  	11 chars
+*/
+var WorkspaceIDPattern = regexp.MustCompile(`^[a-z]{2,16}-[a-z0-9]{2,11}-[a-z0-9]{11}$`)
```

Comment:
I've just noticed that the regex should be `^[a-z0-9-]{3,23}-[a-z0-9]{11}$`, because it always clamp the _username+reponame_ to 23 characters.

Test cases to add:
```
a-b-g6024jgir2j
a-bcdefghijklmnopqrstux-g6024jgir2j
abcdefghijklmnopqrstu-x-g6024jgir2j
```

Reply:
In my regex, i am validating owner name & repo. name separately with a separation of `-`. And, it would work in both the cases. wdyt? 🤔

Reply:
i tried your repo. on the PR preview env. It worked.

Generated Workspace id was `felladrin-pickbetterpac-ronx06hzhja`

Reply:
Exactly, notice that `pickbetterpac` has 13 characters, so it shouldn't pass when using the  rule `[a-z0-9]{2,11}`

Also, it's possible to have numbers in the username too. So it shouldn't pass when using the rule `a-z]{2,16}`. Examples:
- `https://github.com/ritz078`
- `https://github.com/33gustavo33`
- `https://github.com/end3r`

Reply:
resolved in [`4c4292d`](https://github.com/gitpod-io/gitpod/pull/15539/commits/4c4292d02280cfc7d7124b31cd6473077cd9567a)